eftest/src/eftest/runner.clj

```diff
@@ -18,15 +19,23 @@
 (defn- test-vars [ns vars opts]
   (let [once-fixtures (-> ns meta ::test/once-fixtures test/join-fixtures)
         each-fixtures (-> ns meta ::test/each-fixtures test/join-fixtures)
+        capture-context (capture/init-capture)
         report        (synchronize test/report)
-        test-var      (fn [v] (binding [test/report report] (test/test-var v)))]
+        test-var      (fn [v]
+                        (capture/begin-capture)
+                        (binding [test/report report
+                                  test/*test-out* *out*
+                                  *out* (:captured-writer capture-context)
+                                  *err* (:captured-writer capture-context)]
+                          (test/test-var v)))]
     (once-fixtures
      (fn []
        (if (:multithread? opts true)
          (let [test (bound-fn [v] (each-fixtures #(test-var v)))]
            (dorun (->> vars (filter synchronized?) (map test)))
            (dorun (->> vars (remove synchronized?) (pmap test))))
-         (doseq [v vars] (each-fixtures #(test-var v))))))))
+         (doseq [v vars] (each-fixtures #(test-var v))))))
+    (capture/restore-capture capture-context)))
```

Comment:
Can we put this in a `try`-`finally` somehow?

Reply:
Propably. I also considered to put it into the `test-ns` function. The problem is that the context is required in here https://github.com/jarppe/eftest/blob/master/eftest/src/eftest/runner.clj#L28-L29

I'll think about this.